src/compiler/utilities.ts

```diff
@@ -6967,6 +6967,20 @@ namespace ts {
         return compilerOptions[flag] === undefined ? !!compilerOptions.strict : !!compilerOptions[flag];
     }
 
+    export function compilerOptionsAffectSemanticDiagnostics(newOptions: CompilerOptions, oldOptions: CompilerOptions) {
+        if (oldOptions === newOptions) {
+            return false;
+        }
+
+        for (const option of optionDeclarations) {

```

Comment:
`return forEach(optionDeclarations, option => return (option.stringFlag && ...` reads a little better to me.

Reply:
`some` over `forEach`, probably.